make JALR more readable (#880)
riscv/riscv-cheri: src/cheri/rvy-cheriot-isa-unpriv.adoc

{cheriot_unpriv_ext_name} assumes the presence of
both the <<section_zylevels1>> and <<section_zyseal>> extensions.
The present specification presumes the *absence*

> of <<section_cheri_hybrid_ext,{cheri_default_ext_name}>> and

Author: ```suggestion
of  both the <<section_cheri_hybrid_ext,{cheri_default_ext_name}>> and
```

> of  both the <<section_cheri_hybrid_ext,{cheri_default_ext_name}>> and

Author: ```suggestion
of both the <<section_cheri_hybrid_ext,{cheri_default_ext_name}>> and
```

[NOTE]
=====
{cheriot_unpriv_ext_name} is incompatible with

> <<section_rvy_sentry_insn_ext>> as <<sec_cap_type>> of `1` is not a sentry type.

Reviewer: Zycheriot (specifically, §14.1.2 "Capability Types" of the present riscv-cheri doc) _does_ designate otype `1` as a sentry type (which, again, the base defines as "JALR unseals this type"; see §2.3.9).  In fact, it's even _almost_ "the same" sentry as in the "default" encoding's implicit semantics (specifically, it's our forward-arc interrupt-posture-inheriting sentry type).  That said, there are some significant differences: CHERIoT does not use "HOOK 2" to have JALR _produce_ capabilities with type `1` (unlike the default's, where forward- and backward-arc types aren't distinguished) and CHERIoT's encodings cannot represent type `1` capabilities that do not grant X-permission.

Author: The solution here may well be to stop trying to make JALR generic and simply to have it in an extension with different versions for the different encoding formats.
If we can't work out how to make this work between ourselves we're not going to get it through review.
So maybe Zysentry has one vesion of JALR, and there's another for CHERIoT.
And CHERI implementations _must_ choose a version of JALR.
Then we can all write exactly what we want JALR to do without adding the hooks.

Would that solve it?

Reviewer 2: I would just drop this diff from this PR. The current wording looks fine as is to me.

> <<section_rvy_sentry_insn_ext>> as <<sec_cap_type>> of `1` is not a sentry type.

Author: ```suggestion
While {cheriot_unpriv_ext_name} is nominally compatible with
<<section_rvy_sentry_insn_ext>>,
```
